src/Chainweb/Rosetta/RestAPI.hs

```diff
@@ -46,46 +66,166 @@ import Chainweb.Version
 
 ---
 
-type RosettaApi_ =
-    -- Accounts --
-    "rosetta" :> "account" :> "balance"
-        :> ReqBody '[JSON] AccountBalanceReq
-        :> Post '[JSON] AccountBalanceResp
-    -- Blocks --
-    :<|> "rosetta" :> "block" :> "transaction"
-        :> ReqBody '[JSON] BlockTransactionReq
-        :> Post '[JSON] BlockTransactionResp
-    :<|> "rosetta" :> "block"
-        :> ReqBody '[JSON] BlockReq
-        :> Post '[JSON] BlockResp
-    -- Construction --
-    :<|> "rosetta" :> "construction" :> "metadata"
-        :> ReqBody '[JSON] ConstructionMetadataReq
-        :> Post '[JSON] ConstructionMetadataResp
-    :<|> "rosetta" :> "construction" :> "submit"
-        :> ReqBody '[JSON] ConstructionSubmitReq
-        :> Post '[JSON] ConstructionSubmitResp
-    -- Mempool --
-    :<|> "rosetta" :> "mempool" :> "transaction"
-        :> ReqBody '[JSON] MempoolTransactionReq
-        :> Post '[JSON] MempoolTransactionResp
-    :<|> "rosetta" :> "mempool"
-        :> ReqBody '[JSON] MempoolReq
-        :> Post '[JSON] MempoolResp
-    -- Network --
-    :<|> "rosetta" :> "network" :> "list"
-        :> ReqBody '[JSON] MetadataReq
-        :> Post '[JSON] NetworkListResp
-    :<|> "rosetta" :> "network" :> "options"
-        :> ReqBody '[JSON] NetworkReq
-        :> Post '[JSON] NetworkOptionsResp
-    :<|> "rosetta" :> "network" :> "status"
-        :> ReqBody '[JSON] NetworkReq
-        :> Post '[JSON] NetworkStatusResp
+-- ------------------------------------------------------------------ --
+-- Rosetta Api
 
 type RosettaApi (v :: ChainwebVersionT) = 'ChainwebEndpoint v :> Reassoc RosettaApi_
 
--- TODO: Investigate if Rosetta Erros can be dynamic?
+type RosettaApi_ = "rosetta" :>
+    ( -- Accounts --
+      RosettaAccountBalanceApi_
+      -- Blocks --
+    :<|> RosettaBlockTransactionApi_
+    :<|> RosettaBlockApi_
+      -- Construction --
+    :<|> RosettaConstructionMetadataApi_
+    :<|> RosettaConstructionSubmitApi_
+      -- Mempool --
+    :<|> RosettaMempoolTransactionApi_
+    :<|> RosettaMempoolApi_
+      -- Network --
+    :<|> RosettaNetworkListApi_
+    :<|> RosettaNetworkOptionsApi_
+    :<|> RosettaNetworkStatusApi_
+    )
```

Comment:
These changes break up the servant API definition into reusable chunks so that we can generate standalone clients for testing. Then, we recombine and reassociate for the prod api